src/GraphQL/CodeAnalysis/AllowedOnAttribute.cs

```diff
@@ -0,0 +1,15 @@
+using GraphQL.Types;
+
+namespace GraphQL;
+
+/// <summary>
+/// Marker attribute for analyzer usage to specify
+/// graph types where the annotated method can be used.
+/// The generic argument should be an interface.
```

Comment:
Technically, this limitation can be removed. It will be a little bit more expensive to iterate base classes.

Reply:
Seems fine to me.  This is only an internal attribute so it only needs to support the types we plan to use it with.